### ISSUE TITLE
Error messages when migrating custom scalar type (`int64` to `enum`)

### ISSUE THREAD

**Issue by tjjfv** [edgedb/edgedb#5022]
While attempting to create a migration which would change a custom scalar type from `int64` to `enum`, I encountered error messages without detail, in both the interactive mode and with the `--allow-unsafe` option.

Steps to Reproduce:

Attempt to change a custom scalar from `int64` to 
- Start with schema:
```esdl
module default
{
  type Test
  {
    property abc -> CustomScalar;
  }
  scalar type CustomScalar extending int64 {
  }
}
```
- Create & apply migration.
- Modify the schema to:
```esdl
module default
{
  type Test
  {
    property abc -> CustomScalar;
  }
  scalar type CustomScalar extending enum< A, B, C, D, E >;
}
```
- Attempt to create a migration:
  - Using interactive:
  ```
  $ edgedb migration create
  did you alter scalar type 'default::CustomScalar'? [y,n,l,c,b,s,q,?]
  > l
  The following DDL statements will be applied:
      ALTER SCALAR TYPE default::CustomScalar EXTENDING enum<A, B, C, D, E> ;
  did you alter scalar type 'default::CustomScalar'? [y,n,l,c,b,s,q,?]
  > y
  edgedb error: EdgeDB could not resolve migration with the provided answers. Please retry with different answers.
  ```
  - Using allow unsafe:
  ```
  $ edgedb migration create --non-interactive --allow-unsafe 
  edgedb error: EdgeDB could not resolve migration automatically. Please run in interactive mode to confirm changes, or use `--allow-unsafe`
  ```
  Neither error message provides a reason or detail for the error.

Starting with the `enum` and attempting to migrate to the `int64` results in:
```
$ edgedb migration create
edgedb error: SchemaError: cannot change concrete base of a scalar type
```
which provides a reason, but no detail.

- EdgeDB Version: `2.9+a3d80ef`
- EdgeDB CLI Version: `2.9+a3d80ef`
- OS Version: macOS 13.0.1 (22A400)

**Comment by tjjfv** [comment]
Neither 🙂. I'm new to EdgeDB and was/am working on initial modeling on a new project. I decided to do so concurrently with experimenting with the migration process,  to better understand what to expect both in the interactive prompts and the underlying DDL.

Thus, I was repeatedly creating various migrations (possibly less typical scenarios, though not specifically contrived) that arose from drafting the model 'live' in the esdl.

In this scenario, I had hoped for a prompt to map the the int values to the enum values.

**Comment by msullivan** [comment]
I think we probably can't support this, but need to produce good error messages on both sides.
Failing to resolve is pretty bad.
We should probably *not* try to compute an ALTER, since it isn't allowed, and instead suggest a DROP/CREATE pair.
Though unfortunately in this case a DROP/CREATE pair won't help, since it's in use.

**Comment by msullivan** [comment]
I think this was fixed by #5664
